detox/test/e2e/22.launch-args.test.js

```diff
@@ -26,25 +26,55 @@ describe(':android: Launch arguments', () => {
     await assertLaunchArg('heisthe', '1');
   });
 
-  it('should handle complex args', async () => {
+  it('should handle complex args when used on-site', async () => {
     const launchArgs = {
       complex: {
         bull: ['s', 'h', 1, 't'],
         and: {
           then: 'so, me',
         }
       },
-      complexlist: ['arguments', 'https://haxorhost:666'],
+      complexlist: ['arguments', 'https://haxorhost:1337'],
     };
 
     await device.launchApp({newInstance: true, launchArgs});
-
     await element(by.text('Launch Args')).tap();
 
     await assertLaunchArg('complex', JSON.stringify(launchArgs.complex));
     await assertLaunchArg('complexlist', JSON.stringify(launchArgs.complexlist));
   });
 
+  it('should allow for arguments modification', async () => {
+    device.appLaunchArgs.modify({
```

Comment:
Could we add an assertion regarding the original appLaunchArgs values? ... i.e.:

```js
if (!_.isEqual(device.appLaunchArgs.get(), { app: 'le', goo: 'gle', micro: 'soft' }) {
  fail('Please check detox.config.js - your configuration should have custom app launchArgs');
}
```